libraries/AP_HAL_Linux/Perf.cpp

```diff
@@ -186,4 +195,322 @@ void Util::perf_count(perf_counter_t perf)
     perf_counter->count++;
 }
 
-#endif
+#ifdef HAVE_SYSTEMD
+
+#define DBUS_NAMESPACE "org.dronecode.Ardupilot"
+#define DBUS_PATH "/org/dronecode/ardupilot"
+
+PerfManager* PerfManager::_instance = nullptr;
+
+extern const sd_bus_vtable _perf_vtable[];
+
+extern "C" int _dbus_list_perf_callback(sd_bus_message *m, void *data, sd_bus_error *ret_error)
+{
+    PerfManager *perf_manager = (PerfManager *)data;
+    return perf_manager->dbus_list_perf_callback(m, data, ret_error);
+}
+
+extern "C" int _dbus_get_perf_callback(sd_bus_message *m, void *data, sd_bus_error *ret_error)
+{
+    PerfManager *perf_manager = (PerfManager *)data;
+    return perf_manager->dbus_get_perf_callback(m, data, ret_error);
+}
+
+static void* _dbus_thread_callback(void *data)
+{
+    PerfManager *perf_manager = (PerfManager *)data;
+    perf_manager->dbus_process_callback();
+    return nullptr;
+}
```

Comment:
for these 3 functions... usually they are named with "trampoline" since all they do is to call the method within the `void *data` object.